libraries/AP_BoardConfig/AP_BoardConfig.cpp

```diff
@@ -225,6 +226,11 @@ void AP_BoardConfig::init()
     // rebooting
     hal.util->set_imu_target_temp((int8_t *)&_imu_target_temperature);
 #endif
+
+    _rtc_source = rtc.register_rtc_source(AP_RTC::SOURCE_HW);
```

Comment:
this seems to be a huge change in behaviour. With this change it is now trusting the system clock for signing and for reporting the system time, with no way to turn this off. Previously both would be GPS based. The vast majority of boards don't have a usable system clock at boot, so we're just pushing garbage into AP_RTC at boot.